src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -99,41 +106,75 @@ namespace {
 {
-    // TODO: Implement AddThreadNetworkCommand after OCTET_STRING is supported.
-    return EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_NETWORK_NOT_FOUND;
+    EmberAfNetworkCommissioningError err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED;
+
+    static_assert(std::is_same<std::remove_cv_t<decltype(kMaxNetworkIDLen)>, uint8_t>::value);
+    static_assert(sizeof(kTemporaryThreadNetworkId) <= kMaxNetworkIDLen);
+    static_assert(sizeof(kTemporaryThreadNetworkId) <= sizeof(ThreadNetworkInfo::mDataset));
+    VerifyOrExit(operationalDatasetLen <= kMaxThreadDatasetLen, err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
+
+    for (size_t i = 0; i < kMaxNetworks; i++)
+    {
+        if (sNetworks[i].mNetworkType == NetworkType::kUndefined)
+        {
+            memcpy(sNetworks[i].mData.mThread.mDataset, operationalDataset, operationalDatasetLen);
+            // We have verified that the operationalDatasetLen is less than kMaxThreadDatasetLen.
+            sNetworks[i].mData.mThread.mDatasetLen = static_cast<uint8_t>(operationalDatasetLen);
+
+            // A 64bit id for thread networks, currently, we are missing some API for getting the thread network extpanid from the
+            // dataset tlv.
+            // TODO: Set thread network id to real ext panid from dataset tlv.
+            memcpy(sNetworks[i].mNetworkID, kTemporaryThreadNetworkId, sizeof(kTemporaryThreadNetworkId));
+            sNetworks[i].mNetworkIDLen = sizeof(kTemporaryThreadNetworkId);
+
+            sNetworks[i].mNetworkType = NetworkType::kThread;
+            sNetworks[i].mEnabled     = false;
+
+            err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_SUCCESS;
+            break;
+        }
+    }
+
+exit:
+    // TODO: We should encode response command here.
+
+    ChipLogDetail(Zcl, "AddThreadNetwork: %d", err);
+    return err;
 }
 
-// TODO: We cannot get length of ssid / credentials here, should be fixed after we have byte string type in ZCL codegen.
-// Note: SSID and credentials are not necessarily to be string.
 EmberAfNetworkCommissioningError OnAddWiFiNetworkCommandCallbackInternal(app::Command *, EndpointId, const uint8_t * ssid,
-                                                                         const uint8_t * credentials, uint64_t breadcrumb,
+                                                                         size_t ssidLen, const uint8_t * credentials,
+                                                                         size_t credentialsLen, uint64_t breadcrumb,
                                                                          uint32_t timeoutMs)
 {
     EmberAfNetworkCommissioningError err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED;
 
+    // Checks for const values used in this function.
+    static_assert(std::is_same<std::remove_cv_t<decltype(kMaxWiFiSSIDLen)>, uint8_t>::value);
+    static_assert(std::is_same<std::remove_cv_t<decltype(kMaxWiFiCredentialsLen)>, uint8_t>::value);
```

Comment:
Again, move next to the lines of code that depend on the assert, and add a message.